pkg/kapp/clusterapply/exists_change.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package clusterapply
+
+import (
+	"errors"
+
+	ctldiff "github.com/k14s/kapp/pkg/kapp/diff"
+	ctlres "github.com/k14s/kapp/pkg/kapp/resources"
+)
+
+type ExistsChange struct {
+	change              ctldiff.Change
+	identifiedResources ctlres.IdentifiedResources
+}
+
+func (c ExistsChange) ApplyStrategy() (ApplyStrategy, error) {
+	res := c.change.NewResource()
+	return WaitStrategy{res, c}, nil
+}
+
+type WaitStrategy struct {
+	res ctlres.Resource
+	e   ExistsChange
+}
+
+func (c WaitStrategy) Op() ClusterChangeApplyStrategyOp { return "" }
+
+func (c WaitStrategy) Apply() error {
+	exists, _ := c.e.identifiedResources.Exists(c.res, ctlres.ExistsOpts{})
```

Comment:
I think we should catch the error and only if the error contains "Resource doesnt exist", then we send errors.New("Placeholder Resource doesn't exist"). Otherwise, we should send back the original error.

Reply:
In most of the scenarios [here](https://github.com/vmware-tanzu/carvel-kapp/blob/5ce3839d0046d649f4bf915ba5d6fb9ed7f29168/pkg/kapp/resources/resources.go#L372), we are not returning any error. So, should I check for `nil` instead?

Reply:
IMO, it shouldnt rely on what underlying function is considering as error today.. If underlying function tomorrow starts sending more error messages, then it becomes regression issue.

Reply:
Something like 
```golang
if err != nil {
    return err
}
return errors.New("Placeholder resource doesn't exists")
```
Will cover all scenarios in my opinion. Thoughts?

Reply:
@rohitagg2020 The underlying functions send nil when the resource doesn't exists and value of exists as false. So, if we are checking for an error message containing "Resource doesn't exists", then we might not be retrying in any scenario.

Reply:
@100mik Yep, that's what I am thinking.

Reply:
Done.